<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>898</width>
    <height>628</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Editor</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #333333; /* Dark grey background color */
color: rgb(144, 144, 144);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">background-color: #1F1F1F; /* Darker grey background color */
</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>60</y>
      <width>721</width>
      <height>471</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGraphicsView" name="graphicsView">
       <property name="cursor" stdset="0">
        <cursorShape>CrossCursor</cursorShape>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">border: none;
background-color: #1F1F1F; /* Darker grey background color */
</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>10</y>
      <width>481</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="newFileButton">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Open Image</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/open.png</normaloff>:/icons/icons/open.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="undoButton">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Undo</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/undo.png</normaloff>:/icons/icons/undo.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="redoButton">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Redo</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/redo.png</normaloff>:/icons/icons/redo.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="rotateClock">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Rotate Clockwise</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/rotateClock.png</normaloff>:/icons/icons/rotateClock.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>25</width>
       <height>25</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="rotateAnti">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Rotate Anti-Clockwise</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/rotateAnti.png</normaloff>:/icons/icons/rotateAnti.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>25</width>
       <height>25</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="zoomInButton">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Zoom In</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/zoomIn.png</normaloff>:/icons/icons/zoomIn.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>25</width>
       <height>25</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="zoomOutButton">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Zoom Out</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/zoomOut.png</normaloff>:/icons/icons/zoomOut.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>25</width>
       <height>25</height>
      </size>
     </property>
    </widget>
    <widget class="QPushButton" name="cropButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>0</y>
       <width>51</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="toolTip">
      <string>Crop</string>
     </property>
     <property name="toolTipDuration">
      <number>0</number>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/icons/crop.png</normaloff>:/icons/icons/crop.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>30</width>
       <height>30</height>
      </size>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>898</width>
     <height>32</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="styleSheet">
    <string notr="true">font-family: Arial;
font-size: 16px;
padding: 5px; /* Increase the padding to make the menubar bigger */
</string>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="actionCrop"/>
    <addaction name="actionResize"/>
    <addaction name="actionAdd_Text"/>
   </widget>
   <widget class="QMenu" name="menuFilters">
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="title">
     <string>Filters</string>
    </property>
    <addaction name="actionSharpen"/>
    <addaction name="actionBlur"/>
    <addaction name="actionThreshold"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuRotation_2">
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="title">
      <string>Rotation</string>
     </property>
     <addaction name="actionRotate_Clockwise"/>
     <addaction name="actionRotate_Anti"/>
    </widget>
    <widget class="QMenu" name="menuZoom">
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="title">
      <string>Zoom</string>
     </property>
     <addaction name="actionZoom_In"/>
     <addaction name="actionZoom_Out"/>
    </widget>
    <addaction name="menuRotation_2"/>
    <addaction name="menuZoom"/>
   </widget>
   <widget class="QMenu" name="menuColor">
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>Colors</string>
    </property>
    <addaction name="actionBrightness"/>
    <addaction name="actionContrast"/>
    <addaction name="actionGray_Scale"/>
    <addaction name="actionWarming"/>
    <addaction name="actionCooling"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuEdit"/>
   <addaction name="menuColor"/>
   <addaction name="menuFilters"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="styleSheet">
    <string notr="true">background-color: #333333; /* Dark grey background color */
color: #FFFFFF; /* White text color */</string>
   </property>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionCrop">
   <property name="text">
    <string>Crop</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionRotate_Clockwise">
   <property name="text">
    <string>Rotate Clockwise</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>Resize</string>
   </property>
  </action>
  <action name="actionContrast">
   <property name="text">
    <string>Contrast</string>
   </property>
  </action>
  <action name="actionBrightness">
   <property name="text">
    <string>Brightness</string>
   </property>
  </action>
  <action name="actionSharpen">
   <property name="text">
    <string>Sharpen</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
  <action name="actionRotate_Anti">
   <property name="text">
    <string>Rotate Anti</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as</string>
   </property>
  </action>
  <action name="actionGray_Scale">
   <property name="text">
    <string>Gray Scale</string>
   </property>
  </action>
  <action name="actionWarming">
   <property name="text">
    <string>Warming</string>
   </property>
  </action>
  <action name="actionCooling">
   <property name="text">
    <string>Cooling</string>
   </property>
  </action>
  <action name="actionThreshold">
   <property name="text">
    <string>Threshold</string>
   </property>
  </action>
  <action name="actionAdd_Text">
   <property name="text">
    <string>Add Text</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>newFileButton</sender>
   <signal>clicked()</signal>
   <receiver>actionOpen</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>355</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>355</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoButton</sender>
   <signal>clicked()</signal>
   <receiver>actionUndo</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoButton</sender>
   <signal>clicked()</signal>
   <receiver>actionRedo</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>215</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotateClock</sender>
   <signal>clicked()</signal>
   <receiver>actionRotate_Clockwise</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotateAnti</sender>
   <signal>clicked()</signal>
   <receiver>actionRotate_Anti</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomInButton</sender>
   <signal>clicked()</signal>
   <receiver>actionZoom_In</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>365</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomOutButton</sender>
   <signal>clicked()</signal>
   <receiver>actionZoom_Out</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>415</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cropButton</sender>
   <signal>clicked()</signal>
   <receiver>actionCrop</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
